### PR TITLE
Add options: checkLink, checkBlockQuote, checkEmphasis, checkHeader

### LINKED ISSUE #33: Proposal: Add options to customize filter AST type
Currently, filtered AST types are Link, Image, BlockQuote, Emphasis.
https://github.com/textlint-rule/textlint-rule-prh/blob/dd4bb87109a10ed428eee8baee76925b6396c032/src/textlint-rule-prh.js#L136

I want to add `Header` to the blacklist.
But it cannot set as default behavior. It's not needed in many use cases.
So I propose to support `ignoreSyntax` option.

- Type: `Array<String>` (optional)
- Default values: `[Syntax.Link, Syntax.Image, Syntax.BlockQuote, Syntax.Emphasis]`

How about it?

### PR DESCRIPTION
**Pull request by Leko**
Fixed #33

### CODE REVIEW

File: test/textlintrc-test.js [hunk 1]
```diff
@@ -82,6 +82,61 @@ describe(".textlinrc test", function() {
                 assert(result.messages[1].column === 8);
             });
         });
+        it("should not visit Link, Image, BlockQuote, Emphasis by default", async function() {
```
Comment (azu) [inline test/textlintrc-test.js]: `test/textlintrc-test.js` is legacy style test.
It can be moved to `test/prh-rule-tester-test.js`.

**Comment by Leko** [comment]
@azu Maybe, `Image` is meaningless.
For example, `ＪＱＵＥＲＹ` is valid always (even if `checkImage` is true).

Because `Image` has not any children of `Str`.

markdown: `ＪＱＵＥＲＹ`
ast:

```json
{
  "type": "Image",
  "title": "ＪＱＵＥＲＹ",
  "url": "https://example.com",
  "alt": "ＪＱＵＥＲＹ",
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 1,
      "column": 39
    }
  },
  "range": [
    0,
    39
  ],
  "raw": "![ＪＱＵＥＲＹ](https://example.com \"ＪＱＵＥＲＹ\")"
}
```

So we can remove `checkImage` option and `Image` type from the blacklist, right?

**Comment by azu** [comment]
@Leko You're correct.
Current `Image` handling is incorrect.

We should file it as another issue about `Image` and `Code` support.

File: test/textlintrc-test.js (hunk 1 above)
Comment (Leko) [reply test/textlintrc-test.js]: I moved legacy style tests to test/prh-rule-tester-test.js.

**Comment by azu** [comment]
Release https://github.com/textlint-rule/textlint-rule-prh/releases/tag/5.1.0

**Pull request merged**